### PR TITLE
BO - créer la page BO

### PR DESCRIPTION
**Pull request by MeherJaber-Solixy**

### CODE REVIEW

File: src/Controller/AdminController.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+
+class AdminController extends AbstractController
+{
+    private $serializer;
+
+    public function  __construct( SerializerInterface $serializer)
```
Comment (AltyxSolixyAptyx) [inline src/Controller/AdminController.php]: Il manque le commentaire du constructeur

File: src/Controller/AdminController.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+use App\Entity\Code;
+use App\Service\CodeService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+
+class AdminController extends AbstractController
```
Comment (AltyxSolixyAptyx) [inline src/Controller/AdminController.php]: Commentaire du controller missing

File: src/Controller/CodeController.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+class CodeController extends AbstractController
+{
+    private $serializer;
+
+    public function  __construct( SerializerInterface $serializer)
```
Comment (AltyxSolixyAptyx) [inline src/Controller/CodeController.php]: Same qu'en haut

File: src/Controller/CodeController.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * @Route("/distributedCodes", name="distributed_code", methods={"GET"})
+     */
+    public function findCodes(CodeService $codeService)
+    {
+        $codes = $codeService->findCodeByStatus(false);
+
+        foreach ($codes as $code){
+            $data_array[] = array(
```
Comment (AltyxSolixyAptyx) [inline src/Controller/CodeController.php]: - Déclare ton array hors du foreach
- Utilise le format [ ] plutôt que le old array( ...)

File: src/Controller/IndexController.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Entity\User;
+use App\Entity\Voyage;
+use App\Entity\ListCode;
+
+class IndexController extends AbstractController
```
Comment (AltyxSolixyAptyx) [inline src/Controller/IndexController.php]: Controller inutile au projet

File: src/Controller/UserController.php [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+
+
+class UserController extends AbstractController
+{
+
+    private $serializer;
+
+    public function  __construct( SerializerInterface $serializer)
```
Comment (AltyxSolixyAptyx) [inline src/Controller/UserController.php]: Comment please

File: src/Controller/UserController.php [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+
+
+class UserController extends AbstractController
+{
+
+    private $serializer;
```
Comment (AltyxSolixyAptyx) [inline src/Controller/UserController.php]: Comment please

File: src/Controller/UserController.php [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * @Route("/emails", name="email_extract", methods={"GET"})
+     */
+    public function getEmails(UserService $userService){
+
+        $users = $userService->findAll();
+
+        foreach ($users as $user){
+            $data_array[] = array(
```
Comment (AltyxSolixyAptyx) [inline src/Controller/UserController.php]: Use new array format

File: src/Controller/UserController.php [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        return $response;
+    }
+    /**
+     * @Route("/numVols", name="vol_extract", methods={"GET"})
+     */
+    public function getNVols(UserService $userService){
+
+        $users = $userService->findAll();
+
+        foreach ($users as $user){
```
Comment (AltyxSolixyAptyx) [inline src/Controller/UserController.php]: Same as before, use the new format

File: src/Controller/UserController.php [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+            $data_array[] = array(
```
Comment (AltyxSolixyAptyx) [inline src/Controller/UserController.php]: and again new array format

File: src/Entity/Admin.php [hunk 11]
```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\AdminRepository")
+ */
+class Admin
```
Comment (AltyxSolixyAptyx) [inline src/Entity/Admin.php]: Generate phpdoc as comments

File: templates/security/login.html.twig [hunk 12]
```diff
@@ -0,0 +1,36 @@
+{% extends 'base.html.twig' %}
```
Comment (AltyxSolixyAptyx) [inline templates/security/login.html.twig]: No template has to be defined in BO as @SalimHafsi-Solixy is in charge of the front

File: templates/code/index.html.twig [hunk 13]
```diff
@@ -0,0 +1,20 @@
+{% extends 'base.html.twig' %}
```
Comment (AltyxSolixyAptyx) [inline templates/code/index.html.twig]: No template has to be defined in BO as @SalimHafsi-Solixy is in charge of the front

File: templates/base.html.twig [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Comment (AltyxSolixyAptyx) [inline templates/base.html.twig]: No template has to be defined in BO as @SalimHafsi-Solixy is in charge of the front

File: src/Service/UserService.php [hunk 15]
```diff
@@ -0,0 +1,30 @@
+<?php
+namespace  App\Service;
+
+use App\Entity\User;
+use App\Repository\UserRepository;
+
+
+final class UserService
```
Comment (AltyxSolixyAptyx) [inline src/Service/UserService.php]: Missing comments / No need to set the service as final please

File: src/Service/CodeService.php [hunk 16]
```diff
@@ -0,0 +1,35 @@
+<?php
+namespace  App\Service;
+
+use App\Entity\Code;
+use App\Repository\CodeRepository;
+
+
+final class CodeService
```
Comment (AltyxSolixyAptyx) [inline src/Service/CodeService.php]: Missing comments / No need to set the service as final please

File: src/Security/LoginFormAuthenticator.php [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+namespace App\Security;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\AbstractGuardAuthenticator;
+
+class LoginFormAuthenticator extends AbstractGuardAuthenticator
```
Comment (AltyxSolixyAptyx) [inline src/Security/LoginFormAuthenticator.php]: Useless at the moment

File: src/Repository/UserRepository.php [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+/**
+ * @method User|null find($id, $lockMode = null, $lockVersion = null)
+ * @method User|null findOneBy(array $criteria, array $orderBy = null)
+ * @method User[]    findAll()
+ * @method User[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class UserRepository extends ServiceEntityRepository
```
Comment (AltyxSolixyAptyx) [inline src/Repository/UserRepository.php]: missing comments

File: src/Repository/CodeRepository.php [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+/**
+ * @method Code|null find($id, $lockMode = null, $lockVersion = null)
+ * @method Code|null findOneBy(array $criteria, array $orderBy = null)
+ * @method Code[]    findAll()
+ * @method Code[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class CodeRepository extends ServiceEntityRepository
```
Comment (AltyxSolixyAptyx) [inline src/Repository/CodeRepository.php]: missing comments

File: src/Entity/User.php [hunk 20]
```diff
@@ -0,0 +1,217 @@
...
+namespace App\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="App\Repository\UserRepository")
+ */
+class User
```
Comment (AltyxSolixyAptyx) [inline src/Entity/User.php]: Missing comments

File: src/Controller/AdminController.php (hunk 1 above)
Comment (MeherJaber-Solixy) [reply src/Controller/AdminController.php]: ok c noté

**Pull request merged**